### PR TITLE
[PORT][8.0] pos_order_pricelist_change

### PR DESCRIPTION
**Pull request by legalsylvain**
Port pos_order_pricelist_change:
- Use new api for xml and python file;
- OCA convention; (readme file);

### CODE REVIEW

File: pos_order_pricelist_change/models/pos_order.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    @api.multi
+    def action_recompute_pricelist(self):
+        for po in self:
+            for pol in po.lines:
+                res = pol.onchange_product_id(
+                    po.pricelist_id.id, pol.product_id.id, qty=pol.qty,
+                    partner_id=po.partner_id.id)
+                if res['value']['price_unit'] != pol.price_unit:
+                    pol.write(res['value'])
```
Comment (pedrobaeza) [inline pos_order_pricelist_change/models/pos_order.py]: I think you should write always, because it can be a change in another field, not in price_unit

**Comment by pedrobaeza** [comment]
Just out of curiosity, why the bug in the icon?

:+1: except for a minor remark

**Comment by legalsylvain** [comment]
Hi @pedrobaeza. Thanks for your review. 
- the bug is just because this module "fixes", adding on pos order a native behaviour in sale order. For me, not warn the user is like a "bug"; (but I can remove it, no problem);

File: pos_order_pricelist_change/models/pos_order.py (hunk 1 above)
Comment (legalsylvain) [inline pos_order_pricelist_change/models/pos_order.py]: Good catch. Thanks.

**Comment by pedrobaeza** [comment]
Well, the icon is not a major problem

**Comment by legalsylvain** [comment]
@pedrobaeza . Thanks for your review : Done.

**Comment by pedrobaeza** [comment]
:+1:

**Comment by rafaelbn** [comment]
@legalsylvain this module fails when opening a new pos sesion or resuming an existing one.
This is teh error:

```
Odoo Client Error
too much recursion
http://3113858-40-a4731a.runbot2.odoo-community.org/web/js/web.assets_backend/958fc53:4520
```

**Comment by legalsylvain** [comment]
1/ It's not related to my PR ;
2/ it comes from the current pos_pricelist module ; (note A)
3/ OCA / Web / 8.0 is so broken. Quite annoying ;
4/ We really have to include basic JS test executed on runbot or / and Travis on each PR (note B), in a module like pos_test that realize at least the following scenario : 
- open a session ;
- realize a PoS Sale ;
- close the session ;

I uninstalled pos_pricelist and it worked. (you can make the test on Runbot, but please reinstall the module after)

Maybe it's a incompatibility between pos_pricelist and another module of the OCA / pos repository.

A : @ah-taktik, @antespi  : I personaly don't use this module. Do you have some idea ?
B : @gurneyalex, (other ?) : Do you know which technologies I can use for that ? 

Regards.

**Comment by pedrobaeza** [comment]
About B, PhantomJS is already used in Odoo 8 website tests, so I think that is the most straightforward technology to use. It's also used for some tests of POS itself (https://github.com/odoo/odoo/blob/8.0/addons/point_of_sale/test/test_frontend.py)

**Comment by legalsylvain** [comment]
@pedrobaeza : thanks. 
Is this kind of tests executed by Travis ? Runbot ? Both ?
Regards.

**Comment by pedrobaeza** [comment]
Both, because they are executed via `--test-enable` argument. You can see some of these tests in any Travis build log than contains website modules (for example, the current failing website repo - https://travis-ci.org/OCA/website/jobs/77471421)

**Comment by legalsylvain** [comment]
Thanks a lot !
I have to find time to implement tests on OCA / pos repository.

Kind regards.

**Note**
hewww : nice logs : https://travis-ci.org/OCA/website/jobs/77471421#L2551 ;-)

**Comment by AdilHoumadi** [comment]
Hey,

I couldn’t reproduce your bug. Can you give us something more comprehensive via the debug mode. Start your POS in debug mode and copy past the output here. minified JS want help us!

``` javascript
Odoo Client Error
too much recursion
http://3113858-40-a4731a.runbot2.odoo-community.org/web/js/web.assets_backend/958fc53:4520
```

The issue I see is to update your branch of PR : 8.0_PORT_pos_order_pricelist_change because it is out of date.
Here is how I test your module : I took the last version of OCA/pos then I merge the change which coming from your module to this branch then I create a fresh install and the two modules works as expected.

For testing JS, we have something here related to phantom js but I don’t know how to runt it in travis: 
https://github.com/OCA/pos/blob/8.0/pos_pricelist/test/test.py
https://github.com/OCA/pos/blob/8.0/pos_pricelist/static/src/js/tests.js

CC/ @legalsylvain, @rafaelbn, @pedrobaeza 
Thanks.

**Comment by legalsylvain** [comment]
Hi @ah-taktik, 

Sorry for my previous comment. My analyse was bad. 
I merge last upstream in my PR. Runbot is now OK. Thanks.

@rafaelbn :
By the way : This feature only improve backoffice, not front office. because pricelist are not implemented by default in Front Office. If you want such feature, plese see pos_pricelist module.
I'll update the description of the module.

**Comment by legalsylvain** [comment]
@rafaelbn : could you test again ?
Regards.

**Comment by legalsylvain** [comment]
@rafaelbn : any news ?
Kind regards.

**Comment by legalsylvain** [comment]
@AdilHoumadi , @rafaelbn : any news about tests ?

**Comment by rafaelbn** [comment]
Sorry @legalsylvain , rebuilding runbot for testing!

cc @jcarlosmontoya @JGarcia-Panach

**Comment by legalsylvain** [comment]
This works fine on runbot. http://3149752-40-5d1b65.runbot2.odoo-community.org

could you retry ?

**Comment by rafaelbn** [comment]
Tested in runbot :+1: 

Could you please test this PR @jcarlosmontoya @JGarcia-Panach

**Comment by jcarlosmontoya** [comment]
:+1: works fine, thank you

**Comment by JGarcia-Panach** [comment]
Functional test on runbot:

Creating pos order
step1

With surcharge pricelist (100 + 10 $)
step2

Changing pricelist to public pricelist
step3

Recompute with new price (100$)
step4 after recompute

Everithing works fine :+1:

**Pull request merged**